### DISCUSSION TITLE
Getting an error when running a windows published app on Linux

### DISCUSSION THREAD

**Discussion by dtoyer** [AvaloniaUI/Avalonia#5812] [Q&A]
Hi, I've created a .Net 5 "hello world" type app on Windows and now I am trying to publish it to run on Linux (Ubuntu 20.04) but I am getting the following error:

  **_An assembly specified in the application dependencies manifest (QSS.Clearview.Timeshark.deps.json) was not found:
    package: 'runtimepack.Microsoft.NETCore.App.Runtime.linux-x64', version: '5.0.4'_**

Here are the steps I did:

- Installed both .Net 5 runtime and .Net 5 sdk (I realize I should not need the latter) for Ubuntu 20.04
- Did a .Net publish:  **_dotnet publish -r linux-x64 -o myOutput_**
- Copied the files in the myOutput folder to the Linux machine
- Ran command: **_chmod +x ./QSS.Clearview.Timeshark_**
- Ran app: **_./QSS.Clearview.Timeshark_**
- Received error above

Note: I also tried running as a self contained application: **_dotnet publish -c release -r ubuntu.20.04-x64 -o myOutput --self-contained_**
Doing it this way did not give me the error above but instead I got a "segmentation fault" after a couple of minutes of hang time.

I'm presuming I messed up something with the Linux box. Don't have a ton of knowledge in that area. Am hoping someone can provide some troubleshooting suggestions.

Thanks

**Comment by dtoyer** [comment]
Figured out the initial error. My copy routine was not bringing over the createdump file. I brought that file in but now I am experiencing the same issue when publishing as self contained. The terminal window hangs for a minute or two and then gives a segmentation fault error.

**Comment by kekekeks** [comment]
If you have followed any guide with the word `snap` involved in it, follow this guide to purge it from your system - https://www.kevin-custer.com/blog/disabling-snaps-in-ubuntu-20-04/


Then try running the app again.

**Reply by kekekeks** [reply]
Also try `sudo apt-get install ttf-mscorefonts-installer`

**Reply by dtoyer** [reply]
Thanks for the help. no snaps installed and installing the true type fonts did not resolve either. Am going to find another linux box to try this on. Have been doing this on a hosted ubuntu on Windows 10 via WSL2. Will provide an update if still an issue.
